odoo/addons/base/models/res_currency.py

```diff
@@ -406,14 +405,14 @@ def _compute_rate(self):
     @api.depends('rate', 'name', 'currency_id', 'company_id', 'currency_id.rate_ids.rate')
     @api.depends_context('company')
     def _compute_company_rate(self):
-        last_rate = self.env['res.currency.rate']._get_last_rates_for_companies(self.company_id | self.env.company.root_id)
+        last_rate = self.env['res.currency.rate']._get_last_rates_for_companies((self.company_id | self.env.company.root_id).with_company(self.env.company.root_id))
```

Comment:
I think you might want a `sudo` here instead because you might not have access to the root at all